### PR TITLE
Addons documentation reworked 2

### PR DESCRIPTION
**Pull request by comradekingu**
Needs a bit of checkup

### CODE REVIEW

File: docs/admin/addons.rst [hunk 1]
```diff
@@ -113,17 +109,17 @@ Component discovery
...
 Create one main component least likely to disappear in the future, and others
 will employ :ref:`internal-urls` to it as a VCS configuration, and configure it
 to find all components in it.
 
-The matching is done using regular expressions, where power is a tradeoff for
-complexity in configuration. Some examples for common use cases can be found in
-the addon help section.
+The matching is done using regular expressions, where its power is a trade-off
+for reduced complexity in configuration. Some examples for common use-cases can
+be found in the addon help section.
```
Comment (comradekingu) [inline docs/admin/addons.rst:122]: I can't for the life of me find the addon help section.

File: docs/admin/addons.rst [hunk 2]
```diff
@@ -207,10 +203,10 @@ Statistics generator
 
 Generates a file containing detailed info about the translation status.
 
-You can use Django template in both filename and content, see :ref:`markup`
+You can use a Django template in both filename and content, see :ref:`markup`
```
Comment (comradekingu) [inline docs/admin/addons.rst:206]: one/the/a?

File: docs/admin/addons.rst [hunk 3]
```diff
@@ -499,16 +496,16 @@ Additionally, the following environment variables are available:
...
 
-    Language of currently processed translation (not available for component
-    level hooks).
+    Language of currently processed translation (not available for 
+    component-level hooks).
 
 .. envvar:: WL_PREVIOUS_HEAD
 
-    Previous HEAD on update (available only when running post update hook).
+    Previous HEAD on update (available only when running post-update hook).
```
Comment (comradekingu) [inline docs/admin/addons.rst:508]: after?

File: docs/admin/addons.rst [hunk 4]
```diff
@@ -113,17 +109,17 @@ Component discovery
...
-It is triggered on every VCS update, and otherwise similar to the :djadmin:`import_project`
-management command. This way you can track multiple translation
-components within one VCS.
+Triggered each time the VCS is updated, and otherwise similar to
+the :djadmin:`import_project` management command. This way you can track
+multiple translation components within one VCS.
 
 Create one main component least likely to disappear in the future, and others
 will employ :ref:`internal-urls` to it as a VCS configuration, and configure it
 to find all components in it.
```
Comment (comradekingu) [inline docs/admin/addons.rst:118]: What is "it" here?

File: docs/admin/addons.rst [hunk 5]
```diff
@@ -113,17 +109,12 @@ Component discovery
...
-will employ :ref:`internal-urls` to it as a VCS configuration, and configure it
-to find all components in it.
+Triggered each time the VCS is updated, and otherwise similar to
+the :djadmin:`import_project` management command. This way you can track
+multiple translation components within one VCS.
 
-The matching is done using regular expressions, where power is a tradeoff for
-complexity in configuration. Some examples for common use cases can be found in
-the addon help section.
+Matching is done using powerful regular expressions, reducing configuration
```
Comment (comradekingu) [inline docs/admin/addons.rst:116]: configuration complexity of the UI? Why mention its relation to configuration complexity?

File: docs/admin/addons.rst [hunk 6]
```diff
@@ -133,6 +124,13 @@ from where you can check whether the configuration actually matches your needs:
 .. seealso::
 
     :ref:`markup`
+
```
Comment (comradekingu) [inline docs/admin/addons.rst:127]: Don't know if this is the way to do things, but I had great difficulty figuring out what this was, and still don't understand it being where it was …?

**Comment by nijel** [comment]
Just a general remark to this - the first paragraph in each addon description is taken from its description in the code - that one is also shown in Weblate when configuring or installing addons. For example:

https://github.com/WeblateOrg/weblate/blob/032376a196aa0715f996dd97fc8029a47e220823/weblate/addons/generate.py#L66-L69

File: docs/admin/addons.rst (hunk 2 above)
Comment (orangesunny) [reply docs/admin/addons.rst:208]: a is right

File: docs/admin/addons.rst (hunk 3 above)
Comment (orangesunny) [reply docs/admin/addons.rst:510]: yes, after update

File: docs/admin/addons.rst (hunk 5 above)
Comment (orangesunny) [reply docs/admin/addons.rst:116]: If you are not sure about something that much that you ask about that, please don’t change it. At least wait for the answer before the change.

File: docs/admin/addons.rst [hunk 7]
```diff
@@ -499,16 +500,16 @@ Additionally, the following environment variables are available:
...
 
-    Language of currently processed translation (not available for component
-    level hooks).
+    Language of currently processed translation (not available for 
+    component-level hooks).
 
 .. envvar:: WL_PREVIOUS_HEAD
 
-    Previous HEAD on update (available only when running post update hook).
+    Previous HEAD on update (available only when running post-update hook).
```
Comment (comradekingu) [inline docs/admin/addons.rst:512]:
```suggestion
    Previous HEAD after update (only available after running the post-update hook).
```
?

File: docs/admin/addons.rst (hunk 2 above)
Comment (comradekingu) [reply docs/admin/addons.rst:210]:
```suggestion
You can use a Django template for both filename and content, see :ref:`markup`
```
? (assumption is that the Django template isn't in there)

File: docs/admin/addons.rst [hunk 8]
```diff
@@ -113,16 +109,13 @@ Component discovery
...
-will employ :ref:`internal-urls` to it as a VCS configuration, and configure it
-to find all components in it.
+Triggered each time the VCS is updated, and otherwise similar to
+the :djadmin:`import_project` management command. This way you can track
+multiple translation components within one VCS.
 
-The matching is done using regular expressions, where power is a tradeoff for
-complexity in configuration. Some examples for common use cases can be found in
+The matching is done using regular expressions 
+that enables complex configuration, but some knowledge is needed to use them conveniently.
```
Comment (comradekingu) [inline docs/admin/addons.rst:117]:
```suggestion
enabling complex configuration, but some knowledge is required to do so. 
```

File: docs/admin/addons.rst [hunk 9]
```diff
@@ -133,6 +126,13 @@ from where you can check whether the configuration actually matches your needs:
 .. seealso::
 
     :ref:`markup`
+    
+.. hint::
+
+   With :ref:`internal-urls` you can populate the VCS configuration of other components
+   from your main component by pointing to it as weblate://projectname/themaincomponent
```
Comment (nijel) [inline docs/admin/addons.rst:133]: Maybe better to mark that as sample:

```suggestion
   from your main component by pointing to it as ``weblate://projectname/themaincomponent``
```

File: docs/admin/addons.rst [hunk 10]
```diff
@@ -133,6 +126,13 @@ from where you can check whether the configuration actually matches your needs:
...
+   from the :ref:`component-repo` field (in :guilabel:`Manage` ↓ :guilabel:`Settings` ↓
+   :guilabel:`Version control system`) of each resective component.
```
Comment (nijel) [inline docs/admin/addons.rst:135]: Typo?

```suggestion
   :guilabel:`Version control system`) of each respective component.
```

File: docs/admin/addons.rst (hunk 6 above)
Comment (orangesunny) [reply docs/admin/addons.rst:137]: fixed :) Thank you!

**Comment by nijel** [comment]
I've updated the addon descriptions in 68600bbeec4eeeaaac0d4aeba12ed27c91dbab70.

Additionally, I've reverted the change to "base-file" in 1a55077112. It doesn't make sense to change this only in a single location when it's used in the component configuration and other locations.

File: docs/admin/addons.rst [hunk 11]
```diff
@@ -89,14 +87,12 @@ base file.
 Language consistency
 --------------------
 
-Ensures all components within one project have translations for every added
-language for translation.
+Ensures all components within a project have translations for every added
+translated language by creating empty translations in languages that
+have unadded components.
```
Comment (nijel) [inline docs/admin/addons.rst:92]: Feedback by @Fat-Zer on https://hosted.weblate.org/translate/weblate/application/en/?checksum=4deafeb5b1c33e4c:

> The phrase is quite heavy-worded... I suggest to rephrase:
>
>>    Ensures that if a translation to a language will be added to any component of a project an empty translation to that language will be added to all other components of the project as well.

**Pull request merged**